### PR TITLE
chore: move the config into its own component

### PR DESCRIPTION
**Pull request by olevski**
This does a few things:
- reduces (further) the risk that we spawn multiple connection pools (by assigning the connection pool as a class variable on the config class)
- pulls out the `Config` class into its own component - previously it was part of the data services base

### CODE REVIEW

**Comment by olevski** [comment]
This is the followup from #83 @Panaetius

File: components/renku_data_services/config/config.py [hunk 1]
```diff
@@ -66,25 +77,79 @@ def _oidc_discovery(url: str, realm: str) -> Dict[str, Any]:
...
+class DBConfig:
+    """Database configuration."""
+
+    password: str = field(repr=False)
+    host: str = "localhost"
+    user: str = "renku"
+    port: str = "5432"
+    db_name: str = "renku"
+    _async_engine: ClassVar[AsyncEngine | None] = field(default=None, repr=False, init=False)
+    _async_engine_write_lock: ClassVar[asyncio.Lock] = asyncio.Lock()
```
Comment (Panaetius) [inline components/renku_data_services/config/config.py:90]: this isn't used anywhere

File: components/renku_data_services/migrations/env.py [hunk 2]
```diff
@@ -44,6 +42,8 @@ def run_migrations_offline(target_metadata, sync_sqlalchemy_url: str) -> None:
         with context.begin_transaction():
             context.run_migrations()
 
+    engine.dispose()
```
Comment (Panaetius) [inline components/renku_data_services/migrations/env.py:45]: I thought we didn't need dispose?

**Review by Panaetius — CHANGES_REQUESTED** [review]

File: components/renku_data_services/migrations/env.py (hunk 2 above)
Comment (olevski) [reply components/renku_data_services/migrations/env.py]: Yes I forgot we dont need it.

Comment (olevski) [reply components/renku_data_services/migrations/env.py]: Removed.

File: components/renku_data_services/config/config.py (hunk 1 above)
Comment (olevski) [reply components/renku_data_services/config/config.py]: Removed. I forgot it.

**Review by Panaetius — APPROVED** [review]

**Pull request merged**